tests/specs/events.js

```diff
@@ -0,0 +1,1003 @@
+'use strict'
+
+var test = require('tape')
+var dbFactory = require('../utils/db')
+var waitFor = require('../utils/wait-for')
+
+test('has "on" method', function (t) {
+  t.plan(1)
+
+  var db = dbFactory()
+  var store = db.hoodieApi()
+
+  t.is(typeof store.on, 'function', 'has method')
+})
+
+test('has "one" method', function (t) {
+  t.plan(1)
+
+  var db = dbFactory()
+  var store = db.hoodieApi()
+
+  t.is(typeof store.one, 'function', 'has method')
+})
+
+test('has "off" method', function (t) {
+  t.plan(1)
+
+  var db = dbFactory()
+  var store = db.hoodieApi()
+
+  t.is(typeof store.off, 'function', 'has method')
+})
+
+test('store.on("add") with adding one', function (t) {
+  t.plan(2)
+
+  var db = dbFactory()
+  var store = db.hoodieApi()
+  var addEvents = []
+
+  store.on('add', addEventToArray.bind(null, addEvents))
+
+  store.add({
+    foo: 'bar'
+  })
+
+  .then(waitFor(function () {
+    return addEvents.length
+  }, 1))
+
+  .then(function () {
+    t.is(addEvents.length, 1, 'triggers 1 add event')
+    t.is(addEvents[0].object.foo, 'bar', 'event passes object')
+  })
+})
+
+test('store.on("add") with adding two', function (t) {
+  t.plan(3)
+
+  var db = dbFactory()
+  var store = db.hoodieApi()
+  var addEvents = []
+
+  store.on('add', addEventToArray.bind(null, addEvents))
+
+  store.add([
+    {foo: 'bar'},
+    {foo: 'baz'}
+  ])
+
+  .then(waitFor(function () {
+    return addEvents.length
+  }, 2))
+
+  .then(function () {
+    t.is(addEvents.length, 2, 'triggers 2 add event')
+    t.is(addEvents[0].object.foo, 'bar', '1st event passes object')
```

Comment:
We can't be for sure the order of the added items won't change.
Could you please store the Array in a var and sort it, like here:
https://github.com/hoodiehq/pouchdb-hoodie-sync/blob/master/tests/specs/sync.js#L55

Thank you!